### PR TITLE
rename bvk to cv_sum

### LINKED ISSUE #79: Rename bvk in SaplingProvingContext and SaplingVerifyingContext
The `bvk` field in the `SaplingProvingContext` and `SaplingVerifyingContext` structs does not actually contain `bvk`. It actually tracks the sum of the spend `cv` minus the sum of the output `cv`, which is then used at the end along with `valueBalance` to calculate `bvk`. We should rename it (e.g. to `cv_sum`) and document it to make this clearer.

### PR DESCRIPTION
**Pull request by gregdhill**
Signed-off-by: Gregory Hill <gregorydhill@outlook.com>

Closes #79

### CODE REVIEW

File: zcash_proofs/src/sapling/prover.rs [hunk 1]
```diff
@@ -261,7 +262,7 @@ impl SaplingProvingContext {
         let bvk = PublicKey::from_private(&bsk, FixedGenerators::ValueCommitmentRandomness, params);
 
         // In order to check internal consistency, let's use the accumulated value
-        // commitments (as the verifier would) and apply valuebalance to compare
+        // commitments (as the verifier would) and apply value balance to compare
```
Comment (str4d) [inline zcash_proofs/src/sapling/prover.rs]: This is referring top a specific field name:
```suggestion
        // commitments (as the verifier would) and apply value_balance to compare
```

File: zcash_proofs/src/sapling/prover.rs [hunk 2]
```diff
@@ -22,15 +22,16 @@ use crate::circuit::sapling::{Output, Spend};
 /// A context object for creating the Sapling components of a Zcash transaction.
 pub struct SaplingProvingContext {
     bsk: Fs,
-    bvk: edwards::Point<Bls12, Unknown>,
+    // sum of the spend
```
Comment (str4d) [inline zcash_proofs/src/sapling/prover.rs]:
```suggestion
    /// (sum of the Spend value commitments) - (sum of the Output value commitments)
```

File: zcash_proofs/src/sapling/verifier.rs [hunk 3]
```diff
@@ -18,14 +18,15 @@ fn is_small_order<Order>(p: &edwards::Point<Bls12, Order>, params: &JubjubBls12)
 
 /// A context object for verifying the Sapling components of a Zcash transaction.
 pub struct SaplingVerificationContext {
-    bvk: edwards::Point<Bls12, Unknown>,
+    // sum of the spend
```
Comment (str4d) [inline zcash_proofs/src/sapling/verifier.rs]:
```suggestion
    /// (sum of the Spend value commitments) - (sum of the Output value commitments)
```

**Comment by daira** [comment]
Post-hoc utACK

**Pull request merged**